collector_buffered.go

```diff
@@ -27,7 +27,6 @@ func NewBufferedCollector(ctx context.Context, size int, coll Collector) Collect
 		for {
 			select {
 			case <-ctx.Done():
```

Comment:
We can remove the select statement logic here by calling `close(c.pipe)` after L47 below and replace the for loop with:
```
for in := c.pipe {
        c.catcher.Add(c.Collector.Add(in))
}
```

Reply:
I didn't want to do that in this change since I was just trying to do the minimum to get things working. In this case the reading goroutine shouldn't be closing anything so I got rid of the close statement.